### PR TITLE
retain last state on photo or video mode when user returns to camera screen from preview screen

### PR DESCRIPTION
**Pull request by aman-alfresco**

### CODE REVIEW

File: capture/src/main/kotlin/com/alfresco/capture/CaptureModeSelectorView.kt [hunk 1]
```diff
@@ -77,6 +77,13 @@ class CaptureModeSelectorView(
         onMode?.invoke(modes[position])
     }
 
+    fun retainLastState(position: Int) {
```
Comment (agologan) [inline capture/src/main/kotlin/com/alfresco/capture/CaptureModeSelectorView.kt:80]: This is a bad API, the control is already mode-aware.
Just create a `setMode` the index can be returned by looking for it in the `modes` list.

File: capture/src/main/res/color/capture_mode_text_color.xml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <selector xmlns:android="http://schemas.android.com/apk/res/android">
     <item android:color="?android:attr/textColorPrimary" android:state_activated="true"/>
+    <item android:color="?android:attr/textColorPrimary" android:state_enabled="true"/>
```
Comment (agologan) [inline capture/src/main/res/color/capture_mode_text_color.xml:4]: This is not required, the control assumes selection is done via the `active` state.

File: capture/src/main/kotlin/com/alfresco/capture/CaptureModeSelectorView.kt [hunk 3]
```diff
@@ -42,13 +42,13 @@ class CaptureModeSelectorView(
     }
 
     constructor(context: Context) :
-        this(context, null)
+            this(context, null)
```
Comment (agologan) [inline capture/src/main/kotlin/com/alfresco/capture/CaptureModeSelectorView.kt:45]: Fix indentation here and below.

**Review by agologan — CHANGES_REQUESTED** [review]

File: capture/src/main/kotlin/com/alfresco/capture/CaptureModeSelectorView.kt (hunk 3 above)
Comment (aman-alfresco) [reply capture/src/main/kotlin/com/alfresco/capture/CaptureModeSelectorView.kt]: Done

File: capture/src/main/res/color/capture_mode_text_color.xml (hunk 2 above)
Comment (aman-alfresco) [reply capture/src/main/res/color/capture_mode_text_color.xml:4]: selection is done but text seems to be disabled there

File: capture/src/main/kotlin/com/alfresco/capture/CaptureModeSelectorView.kt (hunk 1 above)
Comment (aman-alfresco) [reply capture/src/main/kotlin/com/alfresco/capture/CaptureModeSelectorView.kt:80]: Done

File: capture/src/main/res/color/capture_mode_text_color.xml (hunk 2 above)
Comment (agologan) [reply capture/src/main/res/color/capture_mode_text_color.xml:4]: that just means that there's a bug in the state management and this is an incorrect fix.

Comment (aman-alfresco) [reply capture/src/main/res/color/capture_mode_text_color.xml]: Done

File: capture/src/main/kotlin/com/alfresco/capture/CameraFragment.kt [hunk 4]
```diff
@@ -72,6 +72,9 @@ class CameraFragment : Fragment(), KeyHandler, MavericksView {
         lifecycleScope.launch {
             updateCameraState()
         }
+
+        layout.modeSelectorView.retainLastState(viewModel.mode)
+        setMode(viewModel.mode)
```
Comment (agologan) [inline capture/src/main/kotlin/com/alfresco/capture/CameraFragment.kt:77]: given that setMode does just one thing, i.e. configure the shutter button i think it's bad use of it.
either group multiple mode related settings or not use it at all

Comment (aman-alfresco) [reply capture/src/main/kotlin/com/alfresco/capture/CameraFragment.kt]: Done

**Review by agologan — APPROVED** [review]

**Pull request merged**